INSTALL.md

```diff
@@ -24,56 +21,142 @@
+ => [stage-1 4/5] COPY app /app                                                                                                                      11.8s
+ => [build 6/8] WORKDIR /app/vehicle-command-main                                                                                                     5.6s
+ => [build 7/8] RUN go get ./...                                                                                                                    162.6s
+ => [build 8/8] RUN go build -o /app/bin ./...                                                                                                     2438.3s
+ => [stage-1 5/5] COPY --from=build /app/bin/tesla-control /usr/bin/                                                                                 10.1s
+ => exporting to image                                                                                                                               16.1s
+ => => exporting layers                                                                                                                              15.8s
+ => => writing image sha256:8032d3fc0fb818244a987ea0691df0c666527e6ac2dafade1604dc37a97998e5                                                          0.1s
+```
+2. When completed, check it's there by issuing `docker images`. You will see somthing like this:
+```
+REPOSITORY                   TAG                 IMAGE ID       CREATED          SIZE
+tesla_ble_mqtt               latest              8032d3fc0fb8   14 minutes ago   34.1MB
+golang                       1.22.4-alpine3.20   ae51de63cd1f   3 weeks ago      227MB
+alpine                       3.20.0              31d13e3b0449   6 weeks ago      6.52MB
+hello-world                  latest              f0c407f2ecb9   4 months ago     8.95kB
+```
+3. Follow steps 1 & 2 exactly as per 'Deploy using Command Line'
+4. You will need to edit docker-compose.yml. Change the line `image: "iainbullock/tesla_ble_mqtt:latest"` to `image: "tesla_ble_mqtt:latest"`
+5. Follow steps 4 to 8 exactly as per 'Deploy using Command Line'
+#### Deploy using Dockerfile via Portainer ####
+Again this is for those who can't (or don't want to) use the pre-built images from Dockerhub
+TODO
 ### Activate the Key via Home Assistant
 1. Go to the Integrations page in Home Assistant and click on the MQTT box.
-2. A new device should appear in the list: `tesla_ble_mqtt`
+2. One new device per vehicle should appear in the list: `Tesla_BLE_LRWXXXXXXXXXXX403`
 3. Access the list of entities linked to the device by clicking on it.
-4. Press the button `Generate Keys`.
-5. Optionally, check the logs in Portainer.
+4. Press the button `Generate Keys`
+5. It is useful to monitor the logs during this process
 
 ### Add the Key to the Vehicle
 
 > [!Important]
-> The RPi must be near the vehicle.
+> The Host Device must be near the vehicle.

```

Comment:
FYI.

RPi4 BLE 5.0 (240m)
RPi3 BLE 4.0 (2-50m) ,depends a lot on the adapter and the device (Tesla).

Gotta wonder if Tesla if 4.0 4.1, 4.2 ?  Newer cars possibly 5.x but Elon at times is cheap on the hardware he uses to save a few $... but knowing this info would be great for us. It's not because I have a 5.0 BLE adapter that I can reach my car 240m away .. definitely not if the car has perhaps 4.1.

I just quickly try to see if I could find the info... cuz of course, if the car has 5.0 (it's possible) then telling people to get a 5.0 adapter would likely help them.

Reply:
I think RPi3 is BLE 4.1 (based on `BCM43438`)
https://www.raspberrypi.com/products/raspberry-pi-3-model-b/ => BLE 4.1
https://www.raspberrypi.com/products/raspberry-pi-3-model-b-plus/ => BLE 4.2

Worth mentionning also Pi Zero W:
https://www.raspberrypi.com/products/raspberry-pi-zero-w/ => BLE 4.1
https://www.raspberrypi.com/products/raspberry-pi-zero-2-w/ => BLE 4.2

Reply:
@epenet Thanx for pointing to a true source and bringing in more models. I admit it was late and did try to cut the corner in my research... whatever page I stumble had 4.0 for the RPi3.
 
I decided to check the hardware... The RPi3 B Bluetooth chip is located on the opposite side of all of the remaining components. In most installations, that's "underneath" and that's usually not ideal for RF transmission. If the RPi is located in the basement, the signal will struggle to get out. Would be better to install it on 2nd floor!

The RPi 3 B+ and newer models all have their Bluetooth chip on the same side of the major components.

Apparently 4.2 is better for connectivity over 4.1 : "BLE 4.2 enables 250% faster and more reliable over-the-air data transmission and 10x more packet capacity." Ref: [Symmetry Electronics](https://www.symmetryelectronics.com/blog/ble-4-1-vs-ble-4-2-new-features-and-advantages/)

Considering Tesla cars chip has BLE 4.2 , I'd hope the RPi 3 B+ or later would be "good enough" but depending on the environment (basement, concrete, stone/bricks walls, plaster walls, etc) and the car on the opposite side of the house, that's likely reducing the range to perhaps ~20-25m - maybe even less if too many walls.

I agree w/ @iainbullock a hardware page might be worth to have or a "(Tesla?) Bluetooth Primer" that goes over this topic.. and lets not forget there are BT adapters that comes w/ an external antenna that usually has RF gain. That too is an option we could mention.